src/components/ClinicFinder.tsx

```diff
@@ -0,0 +1,378 @@
+  coordinatesLat: number;
+  coordinatesLong: number;
+  id: number;
+  name: string;
+  showInMobileApp: boolean;
+  type: ClinicType;
+}
+
+export interface GeoLocation {
+  lat: number;
+  long: number;
+}
+
+export default function ClinicFinder(props: ClinicFinder) {
+  const [clinicServices, setClinicServices] = useState<ClinicService[]>(null);
+  const [activeClinicSiteId, setActiveClinicSiteId] = useState<number>(null);
+  const [hoveredMarker, setHoveredMarker] = useState<number>(null);
+  const [clinicServiceDetails, setClinicServiceDetails] = useState(null);
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [showSearchMarker, setShowSearchMarker] = useState<boolean>(false);
+  const [showItemList, setShowItemList] = useState<boolean>(false);
+  const [userGeoLocation, setUserGeoLocation] = useState<GeoLocation>(null);
+  const [showCurrentPosition, setShowCurrentPosition] = useState<boolean>(false);
+  const [customPosition, setCustomPosition] = useState<GeoLocation>(null);
+  const defaultProps = {
+    center: {
+      lat: props.lat,
+      lng: props.lng,
+    },
+    zoom: 10,
+  };
+
+  console.log('PROPS FROM WIDGET2', props);
+  // Get clinicName services
+  const FetchClinicServices = () => {
+    try {
+      axios
+        .get(`${process.env.REACT_APP_BACKEND_URL}/vet-practices`)
+
+        .then(response => {
+          setClinicServices(response.data);
+        });
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  useEffect(() => {
+    FetchClinicServices();
+    if (window) {
+      setShowItemList(window.innerWidth >= 668);
+    }
+    console.log('geolocation: ', navigator.geolocation);
+    if (navigator.geolocation) {
+      getUserGeoLocation();
+    }
+  }, []);
+
+  // Geo location
+  const getUserGeoLocation = () => {
+    const options = {
+      enableHighAccuracy: true,
+      timeout: 5000,
+      maximumAge: 0,
+    };
+    const success = pos => {
+      const crd = pos.coords;
+      setUserGeoLocation({ lat: crd.latitude, long: crd.longitude });
+    };
+    const error = err => {
+      console.warn(`ERROR(${err.code}): ${err.message}`);
+    };
+    navigator.geolocation.getCurrentPosition(success, error, options);
+  };
+
+  // Get nearest Clinics
+  const distanceBetweenCoordinates = (
+    [lat1, lon1]: number[],
+    [lat2, lon2]: number[],
+  ) => {
+    const toRadian = (angle: number) => (Math.PI / 180) * angle;
+    const distance = (a: number, b: number) => (Math.PI / 180) * (a - b);
+    const RADIUS_OF_EARTH_IN_KM = 6371;
+
+    const dLat = distance(lat2, lat1);
+    const dLon = distance(lon2, lon1);
+
+    lat1 = toRadian(lat1);
+    lat2 = toRadian(lat2);
+
+    // Haversine Formula
+    const a =
+      Math.pow(Math.sin(dLat / 2), 2) +
+      Math.pow(Math.sin(dLon / 2), 2) * Math.cos(lat1) * Math.cos(lat2);
+    const c = 2 * Math.asin(Math.sqrt(a));
+
+    let finalDistance = RADIUS_OF_EARTH_IN_KM * c;
+
+    return finalDistance;
+  };
```

Comment:
Feels like a good candidate to be moved to a utilities folder since this file is quite large.